packages/scene-composer/src/components/panels/__tests__/SceneNodeInspectorPanel.spec.tsx

```diff
@@ -112,4 +109,57 @@ describe('SceneNodeInspectorPanel returns expected elements.', () => {
 
     expect(container).toMatchSnapshot();
   });
+
+  it('disable scale and rotation and render correct overlay section when selected scene node is Tag.', async () => {
+    setFeatureConfig({ [COMPOSER_FEATURES.Overlay]: true });
+    getSceneNodeByRef.mockReturnValue({
+      ...baseNode,
+      components: [
+        {
+          type: KnownComponentType.Tag,
+        },
+      ],
+    });
+
+    const { container } = render(<SceneNodeInspectorPanel />);
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('disable scale and rotation when selected scene node is Annotation.', async () => {
+    setFeatureConfig({ [COMPOSER_FEATURES.Overlay]: true });
+    getSceneNodeByRef.mockReturnValue({
+      ...baseNode,
+      components: [
+        {
+          type: KnownComponentType.DataOverlay,
+          subType: Component.DataOverlaySubType.TextAnnotation,
+        },
+      ],
+    });
+
+    const { container } = render(<SceneNodeInspectorPanel />);
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('disable scale and rotation and render correct overlay section when selected scene node is Tag with Overlay panel.', async () => {
+    setFeatureConfig({ [COMPOSER_FEATURES.Overlay]: true });
+    getSceneNodeByRef.mockReturnValue({
+      ...baseNode,
+      components: [
+        {
+          type: KnownComponentType.Tag,
+        },
+        {
+          type: KnownComponentType.DataOverlay,
+          subType: Component.DataOverlaySubType.OverlayPanel,
+        },
+      ],
+    });
+
+    const { container } = render(<SceneNodeInspectorPanel />);
+
+    expect(container).toMatchSnapshot();
+  });
```

Comment:
Blocker: Assert on something instead of relying solely on snapshots.

Reply:
I don't know if there is a way to assert something here since the main focus of these tests are the `disabled` field of the input is set, and I don't know a way to verify that

Reply:
There is data being mutated somewhere and that data should be able to be asserted on. `container` is going to change. Assert on the state of `container`. Otherwise, how does someone know if the snapshot diff is failing for the reason it should be?

Reply:
added